server/data/incidentTypeConfiguration/formWizard.ts

```diff
@@ -77,41 +82,92 @@ export function generateFields(config: IncidentTypeConfiguration): FormWizard.Fi
-            field: question.id,
-            value: answer.code,
-          },
-        }
       }
-      if (answer.commentRequired) {
-        const fieldName = conditionalFieldName(question, answer, 'comment')
-        fields[fieldName] = {
-          name: fieldName,
-          label: 'Comment',
-          component: 'govukInput',
-          validate: ['required'],
-          dependent: {
-            field: question.id,
-            value: answer.code,
-          },
+      // Add comment/date fields
+      for (const answer of activeAnswers) {
+        if (answer.dateRequired) {
+          const fieldName = conditionalFieldName(question, answer, 'date')
+          fields[fieldName] = {
+            name: fieldName,
+            label: 'Date',
+            component: 'mojDatePicker',
+            validate: ['required', 'ukDate'],
+            dependent: {
+              field: question.id,
+              value: answer.code,
+            },
+          }
+        }
+        if (answer.commentRequired) {
+          const fieldName = conditionalFieldName(question, answer, 'comment')
+          fields[fieldName] = {
+            name: fieldName,
+            label: 'Comment',
+            component: 'govukInput',
+            validate: ['required'],
+            dependent: {
+              field: question.id,
+              value: answer.code,
+            },
+          }
         }
       }
-    }
-  })
+    })
 
   return fields
 }
 
 function conditionalFieldName(question: QuestionConfiguration, answer: AnswerConfiguration, suffix: string): string {
   return `${question.id}-${answer.id}-${suffix}`
 }
+
+/**
+ * Checks if the values submitted by the user match the condition's value
+ *
+ * This is used in the next step condition when multiple values are allowed,
+ * e.g. when field is rendered as a list checkboxes
+ *
+ * For example:
+ *
+ * ```
+ * steps = {
+ *   '/animals': {
+ *      id: '/animals',
+ *      fields: ['animal'],
+ *      next: [
+ *        { value: 'dog', op: checkMultipleValues, next: 'q2', field: 'animal' },
+ *        { value: 'cat', op: checkMultipleValues, next: 'q2', field: 'animal' },
+ *        { value: 'fox', op: checkMultipleValues, next: 'q2', field: 'animal' },
+ *      ]
+ *   }
+ *   '/q2': { ... }
+ * }
+ * ```
+ *
+ * as the field has `multiple: true` the value submitted by the user
+ * would be `string[]`, e.g. `['dog', 'fox']`.
+ *
+ * this function would check if the submitted value contains the value in the
+ * condition, and if that's the case then Form Wizard would move the user to
+ * the step specificed in the `next` property of the next step condition
+ * (`q2` in the example above)
+ *
+ * @param submittedValues values submitted by the user
+ * @param _req request not used
+ * @param _res response not used
+ * @param condition next step condition with value to match
+ * @returns true if the value submitted by the user include `condition.value`
+ */
+export function checkMultipleValues(
```

Comment:
i think this function would be better placed outside of this file as it is not specifically tied to generated question/answer steps

Reply:
it is actually, it's in the specific FormWizard format required in the next conditional steps and it also tied to how we currently generate the steps' next (it will likely change once we group answers together by next question id).

For this reason I don't wanna move this in a separate file or even make it public as it's very tied to the current steps.

Reply:
i still think this function is not tied to anything specific here, it's type is what the form wizard lib expects as a next step condition “op”
```typescript
op: (fieldValue: Value, req: Request, res: Express.Response, con: NextStepCondition) => boolean
```
it wouldn't need any modification to work for checkboxes elsewhere. but whatevs :)